extensions/wikia/Flags/maintenance/MoveNotices.php

```diff
@@ -0,0 +1,465 @@
+				$templates = $flagsExtractor->getAllTemplates();
+
+				$size = sizeof( $templates );
+
+				if ( !$size ) {
+					$this->log .= "[WARNING] No templates found on page $pageName\n";
+				} elseif ( $size > 1 ) {
+					$this->log .= "[WARNING] Found more than one ($size) template $this->templateName on page $pageName\n";
+				}
+
+				if ( $size ) {
+					$this->logTemplatesInfo( $templates );
+
+					$flagsToPages = $this->prepareDataForFlagsToPage( $templates[0]['params'] );
+
+					if ( !$list ) {
+						$flagModel->verifyParamsForAdd( $flagsToPages );
+						$flagModel->addFlagsToPage( $flagsToPages );
+
+						$this->log .= "Adding flags to pages: " . json_encode( $flagsToPages ) ."\n";
+					}
+				}
+
+				fwrite( $this->logFile, $this->log );
+				$this->output( $this->log );
+			}
+
+			$this->log = "Processing template: $this->templateName completed \n";
+			$this->log .= "================================================== \n\n\n";
+
+			fwrite($this->logFile, $this->log);
+			$this->output( $this->log );
+		}
+
+		fclose( $this->logFile );
+
+		$this->output( "Processing completed\n" );
+	}
+
+	/**
+	 * Add flag type
+	 */
+	private function addFlagType( FlagType $flagTypeModel, $flagType ) {
+		$flagTypeModel->verifyParamsForAdd( $flagType );
+		$flagId = $flagTypeModel->addFlagType( $flagType );
+
+		if ( $flagId ) {
+			$this->log .= "Flag ID: $flagId added.\n";
+		} else {
+			$this->log .= "[ERROR] Flag is not added!\n";
+			$this->log .= "================================================== \n\n\n";
+			fwrite( $this->logFile, $this->log );
+			$this->output( $this->log );
+		}
+
+		return $flagId;
+	}
+
+	/**
+	 * Log info about all found templates
+	 */
+	private function logTemplatesInfo( $templates ) {
+		foreach ( $templates as $template ) {
+			$this->log .= "Processing template: " . $template['template'] ."\n";
+
+			if (empty($template['params'])) {
+				$this->log .= "No parameters found\n";
+			} else {
+				$this->log .= "Found parameters: \n";
+
+				foreach( $template['params'] as $name => $value ) {
+					$this->log .= "Parameter $name = $value \n";
+				}
+			}
+		}
+	}
+
+	/**
+	 * Prepare data to add flag type
+	 */
+	private function prepareDataForFlagType( FlagType $flagTypeModel, $data ) {
+		if ( !empty( $data[4] ) ) {
+			$parameters = $this->prepareParametersFromCSV( $data[4] );
+		} else {
+			$parameters = null;
+		}
+
+		$flagType = [
+			'wiki_id' => $this->wikiId,
+			'flag_group' => $flagTypeModel->getFlagGroupId( $data[2] ),
+			'flag_name' => $data[1],
+			'flag_view' => $data[0],
+			'flag_targeting' => $flagTypeModel->getFlagTargetingId( $data[3] ),
+			'flag_params_names' => $parameters
+		];
+
+		return $flagType;
+	}
+
+	/**
+	 * Prepare data to add flags to page
+	 */
+	private function prepareDataForFlagsToPage( $params ) {
+		$flagsToPages = [
+			'wiki_id' => $this->wikiId,
+			'page_id' => $this->pageId,
+			'flags' => [
+				[
+					'flag_type_id' => $this->flagId,
```

Comment:
If it's the right variable, can we change its name to avoid confusing `$flagTypeId` and `$flagId`?